src/test/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplateIT.java

```diff
@@ -47,58 +45,64 @@
 import static org.junit.Assert.assertEquals;
 
 @RunWith(SpringJUnit4ClassRunner.class)
-@PropertySource(value = {"classpath:application.properties"})
+@PropertySource(value = { "classpath:application.properties" })
 public class DocumentDbTemplateIT {
-    private static final Person TEST_PERSON = new Person(TestConstants.ID_1, TestConstants.FIRST_NAME,
-            TestConstants.LAST_NAME, TestConstants.HOBBIES, TestConstants.ADDRESSES);
+    private static final Person TEST_PERSON = new Person(ID_1, FIRST_NAME, LAST_NAME, HOBBIES,
+        ADDRESSES);
 
-    private static final Person TEST_PERSON_2 = new Person(TestConstants.ID_2, TestConstants.NEW_FIRST_NAME,
-            TestConstants.NEW_LAST_NAME, TestConstants.HOBBIES, TestConstants.ADDRESSES);
+    private static final Person TEST_PERSON_2 = new Person(ID_2,
+        NEW_FIRST_NAME,
+        NEW_LAST_NAME, HOBBIES, ADDRESSES);
 
-    private static final Person TEST_PERSON_3 = new Person(TestConstants.ID_3, TestConstants.NEW_FIRST_NAME,
-            TestConstants.NEW_LAST_NAME, TestConstants.HOBBIES, TestConstants.ADDRESSES);
+    private static final Person TEST_PERSON_3 = new Person(ID_3,
+        NEW_FIRST_NAME,
+        NEW_LAST_NAME, HOBBIES, ADDRESSES);
 
     @Value("${cosmosdb.uri}")
     private String documentDbUri;
     @Value("${cosmosdb.key}")
     private String documentDbKey;
 
-    private DocumentDbTemplate dbTemplate;
-    private MappingDocumentDbConverter dbConverter;
-    private DocumentDbMappingContext mappingContext;
-    private DocumentCollection collectionPerson;
-    private DocumentDbEntityInformation<Person, String> personInfo;
-    private String collectionName;
+    private static DocumentDbTemplate dbTemplate;
+    private static DocumentDbEntityInformation<Person, String> personInfo;
+    private static String collectionName;
+
+    private static boolean initialized;
 
     @Autowired
     private ApplicationContext applicationContext;
 
     @Before
     public void setup() throws ClassNotFoundException {
-        final DocumentDBConfig dbConfig = DocumentDBConfig.builder(documentDbUri, documentDbKey, DB_NAME).build();
-        final CosmosDbFactory cosmosDbFactory = new CosmosDbFactory(dbConfig);
-
-        mappingContext = new DocumentDbMappingContext();
-        personInfo = new DocumentDbEntityInformation<>(Person.class);
-        collectionName = personInfo.getCollectionName();
-
-        mappingContext.setInitialEntitySet(new EntityScanner(this.applicationContext).scan(Persistent.class));
-        
-        dbConverter = new MappingDocumentDbConverter(mappingContext, null);
-        dbTemplate = new DocumentDbTemplate(cosmosDbFactory, dbConverter, DB_NAME);
-
-        collectionPerson = dbTemplate.createCollectionIfNotExists(this.personInfo);
+        if (!initialized) {
```

Comment:
Not important though, we could put the initialization part in a @BeforeClass method, in which case we don’t need the initialized flag.

Reply:
Junit mandates @BeforeClass method to be static, which is not possible in our case, as it doesn't work with spring @autowired annotation.